<a href="https://colab.research.google.com/github/UnknownJazz/ADET-Act-1/blob/master/Madlos_IntroToAI_Act2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [4]:
df_heart = pd.read_csv('heart.csv')
df_heart.head()

age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  slope  \
0   63    1   3       145   233    1        0      150      0      2.3      0   
1   37    1   2       130   250    0        1      187      0      3.5      0   
2   41    0   1       130   204    0        0      172      0      1.4      2   
3   56    1   1       120   236    0        1      178      0      0.8      2   
4   57    0   0       120   354    0        1      163      1      0.6      2   

   ca  thal  target  
0   0     1       1  
1   0     2       1  
2   0     2       1  
3   0     2       1  
4   0     2       1

In [5]:
# Change the cholesterol level of above 200 to be 1 (risky amount) and 0 (healthy amount)

df_heart.target.value_counts()

target
1    165
0    138
Name: count, dtype: int64

In [6]:
x = df_heart.drop('target', axis=1)
y = df_heart.target

In [7]:
from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.2, random_state=123)

In [9]:
# Logistric Regression
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, confusion_matrix, classification_report
from sklearn.preprocessing import StandardScaler

scaler = StandardScaler()
x_train_scaled = scaler.fit_transform(x_train)
x_test_scaled = scaler.transform(x_test)

lrc = LogisticRegression(max_iter=200)
lrc.fit(x_train_scaled, y_train)
lrc_pred = lrc.predict(x_test_scaled)

print(f'Logistic Regression Accuracy: {accuracy_score(y_test.values, lrc_pred) * 100}%')

conf_matrix = confusion_matrix(y_test.values, lrc_pred)
classification_rep = classification_report(y_test.values, lrc_pred)

print (f'Confusion Matrix: \n{conf_matrix}')
print (f'Classification Report: \n{classification_rep}')



Logistic Regression Accuracy: 75.40983606557377%
Confusion Matrix: 
[[22  8]
 [ 7 24]]
Classification Report: 
              precision    recall  f1-score   support

           0       0.76      0.73      0.75        30
           1       0.75      0.77      0.76        31

    accuracy                           0.75        61
   macro avg       0.75      0.75      0.75        61
weighted avg       0.75      0.75      0.75        61



In [10]:
# Decision Tree Classifier

from sklearn.tree import DecisionTreeClassifier
from sklearn.metrics import accuracy_score, confusion_matrix, classification_report

dtc = DecisionTreeClassifier(random_state=123).fit(x_train, y_train)
dtc_pred = dtc.predict(x_test)

print(f'Decision Tree Classifier Accuracy: {accuracy_score(y_test.values, dtc_pred) * 100}%')

conf_matrix = confusion_matrix(y_test.values, dtc_pred)
classification_rep = classification_report(y_test.values, dtc_pred)

print (f'Confusion Matrix: \n{conf_matrix}')
print (f'Classification Report: \n{classification_rep}')

Decision Tree Classifier Accuracy: 75.40983606557377%
Confusion Matrix: 
[[23  7]
 [ 8 23]]
Classification Report: 
              precision    recall  f1-score   support

           0       0.74      0.77      0.75        30
           1       0.77      0.74      0.75        31

    accuracy                           0.75        61
   macro avg       0.75      0.75      0.75        61
weighted avg       0.75      0.75      0.75        61



The Logistic Regression performs the same as the Decision Tree Classifier, both with the same accuracy of 75.41%.
